CMakeLists.txt

```diff
@@ -1153,51 +1153,34 @@ if(HPX_WITH_NETWORKING)
   endif()
 
   hpx_option(
-    HPX_WITH_PARCELPORT_GASNET_MPI BOOL "Enable the GASNET-MPI based parcelport." OFF
-    CATEGORY "Parcelport"
-  )
-  if(HPX_WITH_PARCELPORT_GASNET_MPI)
-    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET_MPI)
-  endif()
-
-  hpx_option(
-    HPX_WITH_PARCELPORT_GASNET_SMP BOOL "Enable the GASNET-SMP based parcelport." OFF
-    CATEGORY "Parcelport"
-  )
-  if(HPX_WITH_PARCELPORT_GASNET_SMP)
-    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET_SMP)
-  endif()
-
-  hpx_option(
-    HPX_WITH_PARCELPORT_GASNET_UDP BOOL "Enable the GASNET-UDP based parcelport." OFF
-    CATEGORY "Parcelport"
-  )
-  if(HPX_WITH_PARCELPORT_GASNET_UDP)
-    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET_UDP)
-  endif()
-
-  hpx_option(
-    HPX_WITH_PARCELPORT_GASNET_OFI BOOL "Enable the GASNET-OFI based parcelport." OFF
-    CATEGORY "Parcelport"
-  )
-  if(HPX_WITH_PARCELPORT_GASNET_OFI)
-    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET_OFI)
-  endif()
-
-  hpx_option(
-    HPX_WITH_PARCELPORT_GASNET_UCX BOOL "Enable the GASNET-UCX based parcelport." OFF
-    CATEGORY "Parcelport"
-  )
-  if(HPX_WITH_PARCELPORT_GASNET_UCX)
-    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET_UCX)
-  endif()
-
-  hpx_option(
-    HPX_WITH_PARCELPORT_GASNET_IBV BOOL "Enable the GASNET-IBV based parcelport." OFF
-    CATEGORY "Parcelport"
-  )
-  if(HPX_WITH_PARCELPORT_GASNET_IBV)
-    hpx_add_config_define(HPX_HAVE_PARCELPORT_GASNET_IBV)
+    HPX_WITH_PARCELPORT_GASNET_CONDUIT
+    STRING
+    "Define which conduit to use for the gasnet parcelport"
+    "udp"
+    STRINGS "smp;udp;mpi;ofi;ucx;ibv"
+  )
+
+  if(HPX_WITH_PARCELPORT_GASNET_CONDUIT AND ${HPX_WITH_PARCELPORT_GASNET_CONDUIT} STREQUAL "mpi")
+    hpx_add_config_define(HPX_WITH_PARCELPORT_GASNET_MPI)
```

Comment:
@ct-clmsn sorry for getting back to this repeatedly. For me there is still the question whether we still need the `HPX_WITH_PARCELPORT_GASNET_XYZ` cmake configuration variables. 

Also, do we really need the compile time (preprocessor) variables (is there code that depends on those)? If we still need them, please use `HPX_HAVE_PARCELPORT_GASNET_XYZ` as the names for the compile time variables (we have adopted the convention that the cmake variables will be named `HPX_WITH_FOO` while the corresponding compile time variable should be named `HPX_HAVE_FOO`).

Reply:
@hkaiser yep, I've made fixes to simplify things a bit.